src/bin/sol-fbp-generator/type-store.c

```diff
@@ -796,6 +817,7 @@ type_store_read_from_json(struct type_store *store, struct sol_str_slice input)
         if (!read_type(&decoder, desc)) {
             type_description_fini(desc);
             sol_vector_del_last(&store->types);
+            return false;
```

Comment:
A wild `return false;` appears.
1. Delete
2. Leave it there, it was forgotten to start with!